pkg/testing/integration/program.go

```diff
@@ -1107,7 +1181,31 @@ func (pt *programTester) prepareNodeJSProject(projinfo *engine.Projinfo) error {
 
 // preparePythonProject runs setup necessary to get a Python project ready for `pulumi` commands.
 func (pt *programTester) preparePythonProject(projinfo *engine.Projinfo) error {
-	return nil
+	cwd, _, err := projinfo.GetPwdMain()
+	if err != nil {
+		return err
+	}
+
+	// Create a new Pipenv environment. This bootstraps a new virtual environment containing the version of Python that
+	// we requested. Note that this version of Python is sourced from the machine, so you must first install the version
+	// of Python that you are requesting on the host machine before building a virtualenv for it.
+	pythonVersion := pt.getPythonVersion()
+	if err = pt.runPipenvCommand("pipenv-new", []string{"--python", pythonVersion}, cwd); err != nil {
+		return err
+	}
+
+	// Install the package's dependencies, which Pipenv infers from a requirements.txt file in the root of the project.
+	if err = pt.runPipenvCommand("pipenv-install", []string{"install"}, cwd); err != nil {
+		return err
+	}
+
+	// Install the Python SDK that we are testing into the virtualenv. This is the SDK that we just built.
+	sdkPath, err := pt.getPythonSDKPath()
```

Comment:
I am not sure this is going to be correct in up-stack repositories.  In these repositories, we'd want to install the runtime from PyPI but the package under test in this way, correct?

I think that instead we should probably be using the the `Dependencies` array on ProgramTestOptions in some way (maybe it is a path relative to `GOPATH/src`?), similar to how we use these as a way to signal what to `yarn link`?

Reply:
Ah, you're right. I spend too much time in pulumi/pulumi.

Unfortunately pipenv doesn't have any kind of `yarn link` equivalent, so for each repo we're going to need to tell the integration test where to look for the package to install. I think I can get this worked out, though.